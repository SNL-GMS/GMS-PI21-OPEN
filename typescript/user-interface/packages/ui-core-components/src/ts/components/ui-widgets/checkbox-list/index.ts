import { CheckboxList } from './checkbox-list';
import { SimpleCheckboxList } from './simple-checkbox-list';
import type { CheckboxListEntry, CheckboxListProps, SimpleCheckboxListProps } from './types';

export {
  CheckboxList,
  CheckboxListEntry,
  CheckboxListProps,
  SimpleCheckboxList,
  SimpleCheckboxListProps
};
