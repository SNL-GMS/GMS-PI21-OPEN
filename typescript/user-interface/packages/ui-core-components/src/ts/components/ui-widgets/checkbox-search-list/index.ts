import { CheckboxSearchList } from './checkbox-search-list';
import * as CheckboxSearchListTypes from './types';

export { CheckboxSearchList, CheckboxSearchListTypes };
