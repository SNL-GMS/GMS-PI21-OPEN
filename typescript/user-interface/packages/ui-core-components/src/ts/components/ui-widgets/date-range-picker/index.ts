import { DateRangePicker } from './date-range-picker';
import * as DateRangePickerTypes from './types';

export { DateRangePicker, DateRangePickerTypes };
