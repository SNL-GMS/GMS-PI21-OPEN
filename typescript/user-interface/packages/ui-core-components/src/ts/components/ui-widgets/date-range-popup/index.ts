import { DateRangePopup } from './date-range-popup';
import * as DateRangePopupTypes from './types';

export { DateRangePopup, DateRangePopupTypes };
