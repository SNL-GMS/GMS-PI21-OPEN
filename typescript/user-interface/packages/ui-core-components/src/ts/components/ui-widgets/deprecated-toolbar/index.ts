// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { ToolbarComponent } from './toolbar-component';
import * as DeprecatedToolbarTypes from './types';

export { ToolbarComponent as DeprecatedToolbar, DeprecatedToolbarTypes };
