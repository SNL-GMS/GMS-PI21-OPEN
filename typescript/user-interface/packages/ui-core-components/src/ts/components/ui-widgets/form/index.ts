// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { Form } from './form';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import * as FormTypes from './types';

export { Form, FormTypes };
