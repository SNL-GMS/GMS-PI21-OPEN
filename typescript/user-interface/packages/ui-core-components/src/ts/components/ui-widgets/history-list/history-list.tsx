/* eslint-disable react/destructuring-assignment */
import { Menu, MenuDivider, MenuItem } from '@blueprintjs/core';
import React from 'react';

import type { HistoryListProps } from './types';

// Default length of history list, overridden by props.listLength
const DEFAULT_LIST_LENGTH = 7;

/**
 * Accepts a start and end time as input
 * Protects against cases where startTime > endTime
 * Basically two <TimePicker/>'s put in a div with an enter button
 */

export class HistoryList extends React.Component<HistoryListProps, unknown> {
  private selfRef: HTMLDivElement;

  /**
   * React component lifecycle.
   */
  // eslint-disable-next-line react/sort-comp
  public render(): JSX.Element {
    const listLength = this.props.listLength ? this.props.listLength : DEFAULT_LIST_LENGTH;
    const sortedItems = this.props.items.filter((item, index) => index < listLength);
    sortedItems.sort((a, b) => b.index - a.index);

    return (
      <div
        ref={ref => {
          if (ref) {
            this.selfRef = ref;
          }
        }}
      >
        <Menu>
          {this.props.preferredItems ? (
            <>
              {this.props.preferredItems.map(item => (
                <MenuItem key={item.id} text={item.label} onClick={e => this.onClick(e, item.id)} />
              ))}
              <MenuDivider />
            </>
          ) : null}
          {sortedItems.map(item => (
            <MenuItem key={item.id} text={item.label} onClick={e => this.onClick(e, item.id)} />
          ))}
        </Menu>
      </div>
    );
  }

  public componentDidMount(): void {
    if (this.selfRef) {
      this.selfRef.focus();
    }
  }

  private readonly onClick = (event: React.MouseEvent, itemId: string) => {
    this.props.onSelect(itemId);
  };
}
