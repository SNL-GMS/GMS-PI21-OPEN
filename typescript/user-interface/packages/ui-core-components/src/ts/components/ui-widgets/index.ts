import type { CheckboxListEntry } from './checkbox-list';
import { CheckboxSearchList, CheckboxSearchListTypes } from './checkbox-search-list';
import { CopyContents } from './copy-contents';
import { DateRangePicker, DateRangePickerTypes } from './date-range-picker';
import { DateRangePopup, DateRangePopupTypes } from './date-range-popup';
import { DeprecatedToolbar, DeprecatedToolbarTypes } from './deprecated-toolbar';
import { DropDown } from './drop-down';
import { FilterableOptionList, FilterableOptionListTypes } from './filterable-option-list';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { Form, FormTypes } from './form';
import { HistoryList, HistoryListTypes } from './history-list';
import { IntervalPicker, IntervalPickerTypes } from './interval-picker';
import { LabelValue } from './label-value';
import { LoadingSpinner } from './loading-spinner';
import { PopoverButton } from './popover-button';
import { TextArea } from './text-area';
import { TimePicker, TimePickerTypes } from './time-picker';
import { TitleBar, TitleBarTypes } from './title-bar';
import {
  ButtonGroupToolbarItem,
  ButtonToolbarItem,
  ButtonToolbarItemProps,
  CheckboxDropdownToolbarItem,
  CustomToolbarItem,
  DateRangePickerToolbarItem,
  DropdownToolbarItem,
  IntervalPickerToolbarItem,
  LabelValueToolbarItem,
  LoadingSpinnerToolbarItem,
  NumericInputToolbarItem,
  PopoverButtonToolbarItem,
  SwitchToolbarItem,
  Toolbar,
  ToolbarTypes
} from './toolbar';
import { Tooltip2Wrapper, TooltipWrapper } from './tooltip';
import { Widget, WidgetTypes } from './widgets';

export { CheckboxSearchList, CheckboxSearchListTypes };
export { CheckboxListEntry };
export { CopyContents };
export { SimpleCheckboxList } from './checkbox-list/simple-checkbox-list';
export { DropDown };
export { FilterableOptionList, FilterableOptionListTypes };
export { Form, FormTypes };
export { PopoverButton };
export { HistoryList, HistoryListTypes };
export { TimePicker, TimePickerTypes };
export { DateRangePicker, DateRangePickerTypes };
export { DateRangePopup, DateRangePopupTypes };
export { IntervalPicker, IntervalPickerTypes };
export { LabelValue };
export { LoadingSpinner };
export { TextArea };
export {
  ButtonGroupToolbarItem,
  ButtonToolbarItem,
  ButtonToolbarItemProps,
  CheckboxDropdownToolbarItem,
  CustomToolbarItem,
  DateRangePickerToolbarItem,
  DropdownToolbarItem,
  IntervalPickerToolbarItem,
  LabelValueToolbarItem,
  LoadingSpinnerToolbarItem,
  NumericInputToolbarItem,
  PopoverButtonToolbarItem,
  SwitchToolbarItem,
  Toolbar,
  ToolbarTypes
};
export { DeprecatedToolbar, DeprecatedToolbarTypes };
export { TitleBar, TitleBarTypes };
export { Widget, WidgetTypes };
export { Tooltip2Wrapper, TooltipWrapper };
