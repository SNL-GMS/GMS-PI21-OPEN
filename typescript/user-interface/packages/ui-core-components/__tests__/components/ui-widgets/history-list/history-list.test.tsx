/* eslint-disable react/jsx-props-no-spreading */
import { render } from '@testing-library/react';
import * as Enzyme from 'enzyme';
import * as React from 'react';

import { HistoryList } from '../../../../src/ts/components/ui-widgets/history-list';
import type * as HistoryListTypes from '../../../../src/ts/components/ui-widgets/history-list/types';

const historyListItem: HistoryListTypes.HistoryListItem = {
  index: 0,
  label: 'item',
  id: '1'
};

const props: HistoryListTypes.HistoryListProps = {
  items: [historyListItem],
  preferredItems: [historyListItem],
  listLength: 1,
  onSelect: jest.fn()
};

const props2: HistoryListTypes.HistoryListProps = {
  items: [historyListItem],
  preferredItems: [historyListItem],
  listLength: 1,
  onSelect: jest.fn()
};

describe('History List', () => {
  it('to be defined', () => {
    expect(HistoryList).toBeDefined();
  });

  it('History List renders', () => {
    const { container } = render(<HistoryList {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('History List renders with other props', () => {
    const { container } = render(<HistoryList {...props2} />);
    expect(container).toMatchSnapshot();
  });

  it('History List functions and clicks', () => {
    const wrapper3 = Enzyme.mount(<HistoryList {...props} />);
    const instance: HistoryList = wrapper3.find(HistoryList).instance() as HistoryList;

    const spy = jest.spyOn(instance, 'componentDidMount');
    instance.componentDidMount();
    expect(spy).toHaveBeenCalled();

    const input = wrapper3.find('HistoryList');
    const spy3 = jest.spyOn(instance, 'render');
    input.simulate('click');
    instance.render();
    expect(spy3).toHaveBeenCalled();
  });
});
