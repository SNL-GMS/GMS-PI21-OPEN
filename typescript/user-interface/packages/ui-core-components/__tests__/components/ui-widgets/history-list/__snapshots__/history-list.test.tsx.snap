// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History List History List renders 1`] = `
<div>
  <div>
    <ul
      class="bp4-menu"
    >
      <li
        class=""
      >
        <a
          class="bp4-menu-item bp4-popover-dismiss"
          tabindex="0"
        >
          <div
            class="bp4-fill bp4-text-overflow-ellipsis"
          >
            item
          </div>
        </a>
      </li>
      <li
        class="bp4-menu-divider"
      />
      <li
        class=""
      >
        <a
          class="bp4-menu-item bp4-popover-dismiss"
          tabindex="0"
        >
          <div
            class="bp4-fill bp4-text-overflow-ellipsis"
          >
            item
          </div>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`History List History List renders with other props 1`] = `
<div>
  <div>
    <ul
      class="bp4-menu"
    >
      <li
        class=""
      >
        <a
          class="bp4-menu-item bp4-popover-dismiss"
          tabindex="0"
        >
          <div
            class="bp4-fill bp4-text-overflow-ellipsis"
          >
            item
          </div>
        </a>
      </li>
      <li
        class="bp4-menu-divider"
      />
      <li
        class=""
      >
        <a
          class="bp4-menu-item bp4-popover-dismiss"
          tabindex="0"
        >
          <div
            class="bp4-fill bp4-text-overflow-ellipsis"
          >
            item
          </div>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
